src/NCronJob/Execution/JobExecutor.cs

```diff
@@ -69,11 +69,21 @@ public async Task RunJob(JobRun run, CancellationToken stoppingToken)
         var runContext = new JobExecutionContext(run);
         await ExecuteJob(runContext, job, scope);
     }
+    
+    private IJob ResolveJob(IServiceProvider scopedServiceProvider, JobDefinition definition)
+    {
+        if (definition.Type == typeof(DynamicJobFactory))
+            return dynamicJobFactoryRegistry.GetAndDrainJobInstance(scopedServiceProvider, definition);
+
+        var job = scopedServiceProvider.GetService(definition.Type);
+        if (job != null)
+            return (IJob)job;
 
-    private IJob ResolveJob(IServiceProvider scopedServiceProvider, JobDefinition definition) =>
-        definition.Type == typeof(DynamicJobFactory)
-            ? dynamicJobFactoryRegistry.GetAndDrainJobInstance(scopedServiceProvider, definition)
-            : (IJob)scopedServiceProvider.GetRequiredService(definition.Type);
+        LogUnregisteredJob(definition.Type);
+        job = ActivatorUtilities.CreateInstance(scopedServiceProvider, definition.Type);
```

Comment:
I am still not sure whether or not we want to provide this feature.

This call will only work for parameterless c'tors. But let's keep it for now.

Reply:
You may be thinking of Activator.CreateInstance()

ActivatorUtilities.CreateInstance() (which takes the IServiceProvider) is used internally in the asp.net core framework and is a part of Microsoft.Extensions.DependencyInjection. If the type has multiple constructors then it will choose the longest one (most params).

https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.activatorutilities.createinstance?view=net-8.0

https://github.com/dotnet/docs/issues/31785

Reply:
Ahhh yes I thought of `Activator.CreateInstance`.

Okay - let's roll with that and good that you log a warning.